File: Examples/test-suite/objc/return_const_value_runme.m
#import "return_const_value_proxy.h"

int main(int argc ,char *argv[])
{
NSAutoreleasePool *pool =[[NSAutoreleasePool alloc]init];
Foo_ptr  *ptr =[Foo_ptr  getPtr];
if ([ptr  getVal]  != 17)
NSLog(@"Runtime test1 failed");

ptr=[Foo_ptr getConstPtr];
if ([ptr getVal] != 17)
NSLog(@"Runtime test2 failed");

[pool drain];
return 0;
}

File: Examples/test-suite/lua/return_const_value_runme.lua
require("return_const_value")

local p = return_const_value.Foo_ptr.getPtr()
assert(p:getVal() == 17, "test1" .. p:getVal())
p = return_const_value.Foo_ptr.getConstPtr()
assert(p:getVal() == 17, "test2" .. p:getVal())
